### DISCUSSION TITLE
Generate one openapi spec from multiple maven modules

### DISCUSSION THREAD

**Discussion by mschorsch** [smallrye/smallrye-open-api#1381] [Q&A] [answered]
I have a maven multi module project with different quarkus services. I want to generate one openapi spec for all modules. Is that possible? I could'nt find an aggregate goal or similar in the maven plugin.

**Comment by phillip-kruger** [comment, marked as answer]
I am not sure that is possible out of the box. The best workaround I can think of is to let every Quarkus service dump it's own generated OpenAPI schema to a file (by using [quarkus.smallrye-openapi.store-schema-directory](https://quarkus.io/guides/openapi-swaggerui#quarkus-smallrye-openapi_quarkus.smallrye-openapi.store-schema-directory)

Then write a separate app that combine them. You can use the SmallRye API to do the merge. 

Maybe @MikeEdgar has some more ideas.

**Comment by mschorsch** [comment]
@phillip-kruger Thanks for your quick reply.

Would it be possible to add a new goal `aggregate` to the [maven-plugin](https://github.com/smallrye/smallrye-open-api/blob/main/tools/maven-plugin) that provides this functionality?

This is a feature request :)

**Reply by phillip-kruger** [reply]
Sure! Are you keen to provide a PR?

**Reply by mschorsch** [reply]
Unfortunately I have no time at the moment, too many todos on my todo list. Could you still help me with this?

**Reply by phillip-kruger** [reply]
We can definitely do this, but O do not have capacity at the moment. Let's see what @MikeEdgar says

**Reply by MikeEdgar** [reply]
An `aggregate` goal seems like a good idea to me. We should probably add it to the Gradle plugin as well, but not necessarily at the same time. My capacity is somewhat limited as well. Until it's implemented, you might consider doing what @phillip-kruger suggested (dump one YAML per app) and combine them with something like `yq`.
